tla-assignments/src/main/scala/at/forsyte/apalache/tla/assignments/passes/CoverAnalysisPassImpl.scala

```diff
@@ -0,0 +1,79 @@
+package at.forsyte.apalache.tla.assignments.passes
+
+import java.io.{File, FileWriter, PrintWriter}
+import java.nio.file.Path
+
+import at.forsyte.apalache.infra.passes.{Pass, PassOptions, TlaModuleMixin}
+import at.forsyte.apalache.tla.assignments.{CoverChecker, CoverData, ManualAssignments}
+import at.forsyte.apalache.tla.imp.findBodyOf
+import at.forsyte.apalache.tla.lir.storage.BodyMapFactory
+import at.forsyte.apalache.tla.lir.TlaOperDecl
+import at.forsyte.apalache.tla.lir.transformations.TransformationTracker
+import com.google.inject.Inject
+import com.google.inject.name.Named
+import com.typesafe.scalalogging.LazyLogging
+
+class CoverAnalysisPassImpl @Inject()(options: PassOptions,
+                                      tracker: TransformationTracker,
+                                      @Named("AfterCoverAnalysis") nextPass: Pass with TlaModuleMixin )
+  extends CoverAnalysisPass with LazyLogging {
+
+  override def name: String = "CoverAnalysisPass"
+
+  override def execute(): Boolean = {
+    val inModule = tlaModule.get
+
+    val operDecls = inModule.operDeclarations
+    val varSet = inModule.varDeclarations.map(_.name).toSet
+
+    val bodyMap = BodyMapFactory.makeFromDecls(operDecls)
+
+    val nextName = options.getOrElse("checker", "next", "Next")
+    val nextBody = findBodyOf(nextName, inModule.declarations: _*)
+    // We only check for manual assignments in Next
```

Comment:
Technically, `Init` never has assignments (no primes)

Reply:
We produce `InitPrimed` in one of the first passes

Reply:
Exactly, therefore it has no manual assignments to check for

Reply:
Why cannot we write `x := foo` in `Init`?

Reply:
That's still not an assignment, `x' := foo` is (in `InitPrime`). The real question is, do you want a manual assignment (e.g. of x) in `InitPrime`  to force you to manually assign all instances of `x` in `Next`? I think not.

Reply:
I think, yes. Once we tell the users that they can manually annotate the spec with assignments, it would be hard to explain them why they cannot manually annotate `Init`.

Reply:
I'll add the check if you want, just keep in mind that if anyone ever manually assigns to a variable in `Init` they will have to manually assign all assignments to that same variable in `Next`.